setup.sh

```diff
@@ -1,76 +1,175 @@
-    ZSHRC_BACKUP="${ZSHRC_BACKUP}.backup"
-  done
-  ZSHRC_BACKUP_REL=$(realpath --relative-to="${HOME}" "${ZSHRC_BACKUP}")
-  echo "~/.zshrc exists, moving it to ~/${ZSHRC_BACKUP_REL}"
-  mv ${HOME}/.zshrc ${ZSHRC_BACKUP}
-fi
+# Usage: see -h / usage()
 
-NO_ZSHRC=false
-if [ -f "${HOME}/.zshrc" ]; then
-  NO_ZSHRC=true
-else
-  curl -fsSL https://raw.githubusercontent.com/STAOJ/sta-setup/master/.zshrc > ${HOME}/.zshrc
-fi
+#######################################
+# Adds the specified string to the shell-rc files if it doesn't already exist.
+# This adds the specified string to both .zshrc and .bashrc if they exist.
+# Globals:
+#   HOME
+# Arguments:
+#   $1: the string to be added to .bashrc.
+#######################################
 
-if [ -f "${HOME}/.p10k.zsh" ]; then
-  P10K_BACKUP="${HOME}/.p10k.zsh.backup"
-  while [ -f "${P10K_BACKUP}" ]; do
-    P10K_BACKUP="${P10K_BACKUP}.backup"
-  done
-  P10K_BACKUP_REL=$(realpath --relative-to="${HOME}" "${P10K_BACKUP}")
-  echo "~/.p10k.zsh exists, moving it to ~/${P10K_BACKUP_REL}"
-  mv ${HOME}/.p10k.zsh ${P10K_BACKUP}
-fi
+add_to_rc () {
+    for shell_file in "$HOME/.bashrc" "$HOME/.zshrc"; do
+        if  [ -f "$shell_file" ]; then
+            append_if_not_present "$1" "$shell_file"
+        fi
+    done
+}
 
-NO_P10K=false
-if [ -f "${HOME}/.p10k.zsh" ]; then
-  NO_P10K=true 
-else
-  curl -fsSL https://raw.githubusercontent.com/STAOJ/sta-setup/master/.p10k.zsh > ${HOME}/.p10k.zsh
-fi
+#######################################
+# Adds the given string to the given file if it doesn't already exist.
+# Arguments:
+#   $1: the string to be added to the file.
+#   $2: the path to the file.
+#######################################
 
-if [ "$NO_SHELL" = true ]; then
-  echo "Install successful, you don't seem to be using either bash or zsh so we can't figure out how to change your shell to zsh. Using zsh is not necessary, but it has some nice things installed like syntax highlighting and auto completion"
-fi
+append_if_not_present () {
+grep -qx "$1" "$2" || echo "$1" >> "$2"
+}
 
-if [ "$NO_ZSHRC" = true ]; then
-  echo "~/.zshrc already exists and we are unable to create an backup of it. So the recommended ~/.zshrc was not installed. To install it, consider removing your ~/.zshrc and run \n    curl -fsSL https://raw.githubusercontent.com/STAOJ/sta-setup/master/.zshrc > ${HOME}/.zshrc"
-fi
+#######################################
+# Installs brew under the local user from git.
+# Globals:
+#   HOME
+# Arguments:
+#   None
+#######################################
+
+stasetup::install_brew () {
+    set -e 
+    originaldir=$(pwd)
+    if [ ! -d "$HOME/.linuxbrew" ]; then
+        git clone https://github.com/Homebrew/brew "$HOME/.linuxbrew"
+    fi
+
+    $HOME/.linuxbrew/bin/brew update --force --quiet
+    eval "$($HOME/.linuxbrew/bin/brew shellenv)"
+    brew install --force-bottle binutils
+    brew install --force-bottle gcc
+    cd "$HOMEBREW_PREFIX/bin"
+    ln -sf gcc-11 gcc 
```

Comment:
Why do you want to do `-f`? I am a bit confused.

Reply:
It makes it idempotent - if the symlinks already exist, `ln -s` will throw an error. `ln -sf` will overwrite them silently (which is fine here).

Reply:
See my edit - the symlink would exist if you have already run the script before. Should the script update and the user wants to run it again to get the latest changes (or just forgot that things were already setup on that particular device), it shouldn't error out.

Reply:
An alternative would be to only create the links if they don't exist